fe/fe-core/src/main/java/com/starrocks/planner/HudiScanNode.java

```diff
@@ -0,0 +1,320 @@
+    private TupleDescriptor minMaxTuple;
+
+    public HudiScanNode(PlanNodeId id, TupleDescriptor desc, String planNodeName) {
+        super(id, desc, planNodeName);
+        hudiTable = (HudiTable) desc.getTable();
+    }
+
+    @Override
+    protected String debugString() {
+        MoreObjects.ToStringHelper helper = MoreObjects.toStringHelper(this);
+        helper.addValue(super.debugString());
+        helper.addValue("hudiTable=" + hudiTable.getName());
+        return helper.toString();
+    }
+
+    public void setSelectedPartitionIds(Collection<Long> selectedPartitionIds) {
+        this.selectedPartitionIds = selectedPartitionIds;
+    }
+
+    public void setIdToPartitionKey(Map<Long, PartitionKey> idToPartitionKey) {
+        this.idToPartitionKey = idToPartitionKey;
+    }
+
+    public List<Expr> getNonPartitionConjuncts() {
+        return nonPartitionConjuncts;
+    }
+
+    public List<Expr> getNoEvalPartitionConjuncts() {
+        return noEvalPartitionConjuncts;
+    }
+
+    public List<Expr> getMinMaxConjuncts() {
+        return minMaxConjuncts;
+    }
+
+    public void setMinMaxTuple(TupleDescriptor tuple) {
+        minMaxTuple = tuple;
+    }
+
+    public void getScanRangeLocations(DescriptorTable descTbl) throws UserException {
+        if (selectedPartitionIds.isEmpty()) {
+            return;
+        }
+
+        long start = System.currentTimeMillis();
+        List<PartitionKey> partitionKeys = Lists.newArrayList();
+        List<DescriptorTable.ReferencedPartitionInfo> partitionInfos = Lists.newArrayList();
+        for (long partitionId : selectedPartitionIds) {
+            PartitionKey partitionKey = idToPartitionKey.get(partitionId);
+            partitionKeys.add(partitionKey);
+            partitionInfos.add(new DescriptorTable.ReferencedPartitionInfo(partitionId, partitionKey));
+        }
+        List<HivePartition> hudiPartitions = hudiTable.getPartitions(partitionKeys);
+
+        for (int i = 0; i < hudiPartitions.size(); i++) {
+            descTbl.addReferencedPartitions(hudiTable, partitionInfos.get(i));
+            for (HdfsFileDesc fileDesc : hudiPartitions.get(i).getFiles()) {
+                for (HdfsFileBlockDesc blockDesc : fileDesc.getBlockDescs()) {
+                    addScanRangeLocations(partitionInfos.get(i).getId(), fileDesc, blockDesc,
+                            hudiPartitions.get(i).getFormat());
+                    LOG.debug("Add scan range success. partition: {}, file: {}, block: {}-{}",
+                            hudiPartitions.get(i).getFullPath(), fileDesc.getFileName(), blockDesc.getOffset(),
+                            blockDesc.getLength());
+                }
+            }
+        }
+        LOG.debug("Get {} scan range locations cost: {} ms", result.size(), (System.currentTimeMillis() - start));
+    }
+
+    private void addScanRangeLocations(long partitionId, HdfsFileDesc fileDesc, HdfsFileBlockDesc blockDesc,
```

Comment:
this is repeated with hive table, we should add an abstraction for this like PredicateUtils

Reply:
As the PR describtion said, I will consider these refactor in the following PRs because there are many places that need to be modified, which will interfere review process of the PR.

Reply:
actually i think those method can add some abstraction in this pr @miomiocat